cds/modules/migrator/tasks.py

```diff
@@ -71,3 +83,137 @@ def clean_record(data, source_type):
+            "https://videos.cern.ch/api/mediaexport?id={rn}",
+        ])
+
+    url_info = {'recid': record['recid'], 'rn': record['report_number'][0]}
+
+    for url in urls:
+        url = url.format(url_info)
+        response = requests.get(url)
+        if response.status_code == 401:
+            if is_public(record, 'read'):
+                logger.error('Record {0} should be public in {1}'.format(
+                    record['recid'], url))
+        elif response.status_code != 200:
+            logger.error('Cannot access record {0} via {1}'.format(
+                record['recid'], url))
+        sleep(1)
+
+
+def _check_es(record):
+    """Check if the record is correctly indexed."""
+    index, doc_type = current_record_to_index(record)
+    if not current_search_client.exists(index, doc_type, record.id):
+        logger.error('Record not indexed {0}'.format(record['recid']))
+
+
+def _chec_files(record, recorddump):
+    """Check file integrity."""
+    record_bucket = RecordsBuckets.query.filter(
+        RecordsBuckets.record_id == record.id).one_or_none()
+    if not record_bucket:
+        logger.error('Bucket not found: {0}'.format(recorddump.recid))
+        raise Exception(
+            'Files for record {0} not migrated'.format(recorddump.recid))
+
+    old_files = recorddump.revisions[-1][1].get('_files')
+    # Verify master file
+    try:
+        master_file = [
+            f for f in old_files if f['tags']['context_type'] == 'master'
+        ][0]
+    except IndexError:
+        logger.error('Master file not found: {0}'.format(recorddump.recid))
+        return
+    master_file_path = current_migrator.records_dumploader_cls._get_full_path(
+        master_file['filepath'])
+
+    master_obj = ObjectVersion.get(record_bucket.bucket, master_file['key'])
+    if not master_obj:
+        # Before raising verify the master is accessible on DFS
+        if not (os.path.isfile(master_file_path) and
+                os.access(master_file_path, os.R_OK)):
+            logger.error('Master file found but not migrated: {0}'.format(
+                recorddump.recid))
+            raise Exception('Master file for record {0} not migrated'.format(
+                recorddump.recid))
+        else:
+            logger.warning('Master file found in dump but not on DFS: {0}'.
+                           format(recorddump.recid))
+            return
+    # Verify master object checksum
+    old_master_md5 = _md5(master_file_path)
+    if master_obj.file.checksum != old_master_md5:
+        logger.error(
+            'Master file checksum not correct: {0}'.format(recorddump.recid))
+        raise Exception('Wrong checksum {0}'.format(recorddump.recid))
+
+    # At this point we know the master file is correct, check the other files
+    master_file = CDSVideosFilesIterator.get_master_video_file(record)
+    # Check frames
+    frames = CDSVideosFilesIterator.get_video_frames(master_file)
```

Comment:
can we check that frames AND slaves and any other files exist in EOS, as for the master?

Reply:
This is exactly what this is doing, but just for frames and slaves, I have to check how to do it for other files, like subtitles.